### PR TITLE
feat: prepared statements

### PR DESCRIPTION
**Pull request by barakalon**
Phew, ok.

A few changes:
1. Support for prepared statements
2. Support for setting variables
3. Added mysql-connector-python integration tests, since aiomysql doesn't support prepared statements

This was all done in pursuit of a Tableau integration. Not there yet, though.

### CODE REVIEW

File: mysql_mimic/connection.py [hunk 1]
```diff
@@ -39,50 +54,422 @@ def __init__(
...
+        https://dev.mysql.com/doc/internals/en/com-stmt-fetch.html
+
+        COM_STMT_FETCH fetches rows from an existing resultset after a COM_STMT_EXECUTE.
+        """
+        r = io.BytesIO(data)
+        stmt_id = types.read_uint_4(r)
+        num_rows = types.read_uint_4(r)
+        stmt = self.get_stmt(stmt_id)
+
+        i = 0
```
Comment (tobymao) [inline mysql_mimic/connection.py:284]: this is a bit weird, side effect, you set i = 0, but then have python override i with scope from the for loop, can we make this a bit more explicit, i assume you did this for the null case?

File: mysql_mimic/results.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+
+def bencode(val):
+    for py_type, (_, _, encoder) in _ENCODERS.items():
+        if isinstance(val, py_type):
+            return encoder(val)
+    return _bencode_str(val)
+
+
+def tencode(val):
```
Comment (tobymao) [inline mysql_mimic/results.py:186]: bencode is byte encode, tencode is ?

Comment (barakalon) [reply mysql_mimic/results.py:186]: text encode.
MySQL has "Text result sets" and "binary result sets". So bencode is binary encode.
Let me add more explicit names

File: mysql_mimic/connection.py (hunk 1 above)
Comment (barakalon) [reply mysql_mimic/connection.py:284]: Haha sure

**Pull request merged**